criu/sk-inet.c

```diff
@@ -225,6 +226,31 @@ static int dump_sockaddr(union libsoccr_addr *sa, u32 *pb_port, u32 *pb_addr)
 	return -1;
 }
 
+/*
+ * There is no direct way to get shutdown state for unconnected sockets,
+ * but we can get it indirectly from polling events for a socket.
+ */
```

Comment:
It looks like `ss` can detect `SHUT_RD`, `SHUT_WR`, `SHUT_RDWR` and non-shutdown sockets: https://github.com/shemminger/iproute2/commit/5b816047537bb04620718e798ec09ac9f7a0ad67